### ISSUE TITLE
Cannot build after enabling AndroidX in config.xml

### ISSUE THREAD

**Issue by nijakobius** [skwasjer/skwas-cordova-plugin-datetimepicker#44]
**Describe the bug**
When I add `<preference name="AndroidXEnabled" value="true" />` in `config.xml`, the project doesn't build in Android Studio with the following error message:
```
/Users/USER/Documents/Projekte/APPNAME/App/APPNAME/platforms/android/app/src/main/java/com/skwas/cordova/datetimepicker/TimePickerDialog.java:7: error: package android.support.annotation does not exist
import android.support.annotation.NonNull;
```

AndroidX was recently added with Cordova Android 9.0.0:
https://cordova.apache.org/announcements/2020/06/29/cordova-android-9.0.0.html

- `cordova -v` 10.0.0
- `cordova platform ls` android 9.0.0
- `skwas-cordova-plugin-datetimepicker 2.1.1 "DateTime picker"`
 - OS: macOS X

**Comment by nijakobius** [comment]
Okay, I found the solution:

edit the following files in `src/android`:

**DatePickerDialog.java**
```
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.annotation.StyleRes;
```
Change to
```
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.annotation.StyleRes;
```

**DateTimePicker.java**
```
import android.support.annotation.NonNull;
```
Change to
```
import androidx.annotation.NonNull;
```

**TimePickerDialog.java**
```
import android.support.annotation.NonNull;
```
Change to
```
import androidx.annotation.NonNull;
```

**Comment by skwasjer** [comment]
Refer to this: https://developer.android.com/jetpack/androidx/migrate

Specifically you may try:
```
android.enableJetifier=true
```

I am not very active in Android development so am not up to date on the latest/greatest and when the time is right to update, seeing that there are alot of consumers of this package and I don't want to force people to upgrade Android Studio (albeit 3.2 is fairly old now). So I will leave the issue open for the community to provide input and decide.

**Comment by beinginorbit** [comment]
This doesn't seem to work anymore with Android requiring minTargetSDK of 29.

**Comment by skwasjer** [comment]
@spacesuitdiver try adding https://www.npmjs.com/package/cordova-plugin-androidx-adapter, it seems to work ok for me with SDK 29 and AndroidX preference enabled. It is just a build-time dependency so should not affect the runtime in any way.

**Comment by skwasjer** [comment]
See also [this test suite](https://github.com/skwasjer/skwas-cordova-plugin-datetimepicker/runs/2372126589?check_suite_focus=true) I ran to verify it still works fine. I will update plugin docs accordingly.

**Comment by tullyj** [comment]
With latest cordova-android 10, and Android requiring target api level 30, this is no longer working for me...

**Comment by skwasjer** [comment]
Released under [v3.0.0](https://github.com/skwasjer/skwas-cordova-plugin-datetimepicker/releases/tag/v3.0.0)

Note the [build requirements](https://github.com/skwasjer/skwas-cordova-plugin-datetimepicker#build-requirements). Specifically:

- AndroidX support must be enabled:
  - `cordova-android@8.0.0`  
    Requires additional plugin [`cordova-plugin-androidx`](https://www.npmjs.com/package/cordova-plugin-androidx).
  - `cordova-android@9.0.0`  
    Requires `<preference name="AndroidXEnabled" value="true" />` to be added to `config.xml`.
  - `cordova-android@10.0.0`  
    AndroidX is enabled by default.

Tested against all versions, with different Cordova and NodeJS versions, and minimum Android SDK versions:
https://github.com/skwasjer/skwas-cordova-plugin-datetimepicker/actions/runs/1644162531
